From c1841e6bea92d8b5f184164f012013acf30d0830 Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Wed, 18 Jan 2012 10:38:10 +0100
Subject: [PATCH 16/52] Fix net_check_clients warnings: make it per vlan.

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1326883126-22053-17-git-send-email-armbru@redhat.com>
Patchwork-id: 36583
O-Subject: [RHEL-6.3 PATCH qemu-kvm 16/52] Fix net_check_clients warnings: make it per vlan.
Bugzilla: 758194
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>

From: Tristan Gingold <gingold@adacore.com>

Signed-off-by: Tristan Gingold <gingold@adacore.com>
Reviewed-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>
(cherry picked from commit ac60cc18711a9786af9844d7e3d002276fbd85f3)
Signed-off-by: Markus Armbruster <armbru@redhat.com>
---
 net.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 net.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/net.c b/net.c
index 28103e1..a0a4b90 100644
--- a/net.c
+++ b/net.c
@@ -1363,9 +1363,10 @@ void net_check_clients(void)
 {
     VLANState *vlan;
     VLANClientState *vc;
-    int has_nic = 0, has_host_dev = 0;
 
     QTAILQ_FOREACH(vlan, &vlans, next) {
+        int has_nic = 0, has_host_dev = 0;
+
         QTAILQ_FOREACH(vc, &vlan->clients, next) {
             switch (vc->info->type) {
             case NET_CLIENT_TYPE_NIC:
-- 
1.7.7.5

